src/core/base/basePickBy.js

```diff
@@ -0,0 +1,28 @@
+import baseGet from './baseGet'
+import baseSet from './baseSet'
+import getSize from '../util/getSize'
+import getKey from '../util/getKey'
+import setKey from '../util/setKey'
+import castPath from '../util/castPath'
+import withMutations from '../with/withMutations'
+import { hintConvert, get } from '../'
+
+const setPaths = withMutations((result, data, paths, predicate) => {
+  const length = getSize(paths)
+  let index = -1
+
+  while (++index < length) {
+    const path = get(paths, index)
+    const value = baseGet(data, path, getKey)
+    if (predicate(value, path)) {
```

Comment:
This doesn't seem quite right to me. Shouldn't this be checking to see if `data` has path? I don't quite understand how the tests are passing given this predicate should be failing...